### PR TITLE
[TTIR] Remove TTIR operand_constraints

### PR DESCRIPTION
**Pull request by jserbedzijaTT**
Remove `operand_constraints` from the TTIR dialect and all associated tests. Until a workaround pass is created, we will need to manually insert `to_layout` operations in cases where the default layout of tensors (tiled, DRAM, interleaved) needs to be changed. 

I will remove operand_constraints from the generic op in a separate pull request.

### CODE REVIEW

File: lib/Dialect/TTIR/Transforms/Layout.cpp [hunk 1]
```diff
@@ -176,17 +176,21 @@ createToLayoutOp(PatternRewriter &rewriter, Location loc, Value input,
...
-      getLegalMemorySpace(operandConstraint, defaultMemorySpace);
+  // auto desiredMemorySpace =
+  //     getLegalMemorySpace(operandConstraint, defaultMemorySpace);
 
-  auto desiredMemoryLayout = getLegalTensorMemoryLayout(
-      operandConstraint, desiredMemorySpace, defaultDeviceMemoryLayout);
+  // auto desiredMemoryLayout = getLegalTensorMemoryLayout(
+  //     operandConstraint, desiredMemorySpace, defaultDeviceMemoryLayout);
+
+  // bool tiled =
```
Comment (topke011) [inline lib/Dialect/TTIR/Transforms/Layout.cpp:187]: Remove this

File: lib/Dialect/TTIR/Transforms/Layout.cpp [hunk 2]
```diff
@@ -176,17 +176,21 @@ createToLayoutOp(PatternRewriter &rewriter, Location loc, Value input,
 
 static std::optional<Value>
 createToLayoutOp(PatternRewriter &rewriter, Location loc, Value input,
```
Comment (topke011) [inline lib/Dialect/TTIR/Transforms/Layout.cpp:178]: It seems there is no point to this function anymore, you can remove it and call the latter one.

File: lib/Dialect/TTIR/Transforms/Layout.cpp [hunk 3]
```diff
@@ -223,16 +227,16 @@ class TTIRLayoutDPSOperandsRewriter
       if (mlir::isa<Conv2dOp>(op.getOperation()) && !isResult) {
         continue;
       }
-      auto operandConstraint =
-          mlir::cast<OperandConstraintAttr>(
-              mlir::cast<TTIROp>(op.getOperation())
-                  .getOperandConstraints()[operand.getOperandNumber()])
-              .getValue();
+      // auto operandConstraint =
```
Comment (topke011) [inline lib/Dialect/TTIR/Transforms/Layout.cpp:230]: nit remove

File: lib/Dialect/TTNN/Transforms/TTNNLayout.cpp [hunk 4]
```diff
@@ -248,32 +248,14 @@ createToLayoutOp(PatternRewriter &rewriter, Location loc, Value input,
...
-        utils::toTTNNTensorMemoryLayout(desiredMemoryLayout);
-    ttnnMemoryLayoutAttr =
-        TensorMemoryLayoutAttr::get(rewriter.getContext(), ttnnMemoryLayout);
-  }
-
-  // Check if the tensor should be tiled
-  bool tiled =
-      !bitEnumContainsAny(operandConstraint, OperandConstraint::Scalar);
+static std::optional<Value> createToLayoutOp(PatternRewriter &rewriter,
+                                             Location loc, Value input) {
```
Comment (topke011) [inline lib/Dialect/TTNN/Transforms/TTNNLayout.cpp:252]: Same comment as above, you can remove this helper function.

**Comment by topke011** [comment]
Joco I'm curious did all test passed once you removed all references to operandcontraints from code/tests?

**Review by topke011 — APPROVED** [review]

**Comment by mtopalovicTT** [comment]
Joco I'm curious did all test passed once you removed all references to operandcontraints from code/tests?

**Review by mtopalovicTT — APPROVED** [review]

**Comment by jserbedzijaTT** [comment]
> Joco I'm curious did all test passed once you removed all references to operandcontraints from code/tests?

I still didn't try to run all silicon tests in local, I was hoping CI will do the job for me. Will try today and make an update.

File: lib/Dialect/TTNN/Transforms/TTNNLayout.cpp [hunk 5]
```diff
@@ -334,17 +303,14 @@ class TTNNLayoutDPSOperandsRewriter
...
       Location newLoc =
           appendInputSuffix(op.getLoc(), operand.getOperandNumber());
       // Given the operand constraint, create the desired layout for the operand
-      std::optional<Value> desiredLayout =
-          createToLayoutOp(rewriter, newLoc, operand.get(), operandConstraint);
+      std::optional<Value> desiredLayout = createToLayoutOp(
+          rewriter, newLoc, operand.get(), g_defaultMemorySpaceDevice,
+          TensorMemoryLayoutAttr::get(rewriter.getContext(),
+                                      g_defaultMemoryLayout),
+          true);
```
Comment (mtopalovicTT) [inline lib/Dialect/TTNN/Transforms/TTNNLayout.cpp:313]: nit: `true /* isTiled */)

**Review by vprajapati-tt — APPROVED** [review]
LGTM (in the context of the python test_infra).

**Review by nsmithtt — COMMENTED** [review]
A very welcome change! 🙌

**Review by nsmithtt — APPROVED** [review]

**Review by odjuricicTT — APPROVED** [review]

**Review by nobradovictt — APPROVED** [review]

**Pull request merged**